tests/integration/test_integration_all_rust.py

```diff
@@ -1273,105 +1298,118 @@ def test_big_message(self):
+        await client.hello()
+        result = await client.register(channel_id=chid, key="af1883%&!@#*(", status=400)
         assert result["status"] == 400
 
-        yield self.shut_down(client)
+        await self.shut_down(client)
 
-    @inlineCallbacks
+    @pytest.mark.asyncio
     @max_logs(endpoint=44)
-    def test_msg_limit(self):
+    async def test_msg_limit(self):
         """Test that sent messages that are larger than our size limit are rejected."""
-        client = yield self.quick_register()
+        client = await self.quick_register()
         uaid = client.uaid
-        yield client.disconnect()
+        await client.disconnect()
         for i in range(MSG_LIMIT + 1):
-            yield client.send_notification(status=201)
-        yield client.connect()
-        yield client.hello()
+            await client.send_notification(status=201)
+        await client.connect()
+        await client.hello()
         assert client.uaid == uaid
         for i in range(MSG_LIMIT):
-            result = yield client.get_notification()
+            result = await client.get_notification()
             assert result is not None, f"failed at {i}"
-            yield client.ack(result["channelID"], result["version"])
-        yield client.disconnect()
-        yield client.connect()
-        yield client.hello()
+            await client.ack(result["channelID"], result["version"])
+        await client.disconnect()
+        await client.connect()
+        await client.hello()
         assert client.uaid != uaid
-        yield self.shut_down(client)
+        await self.shut_down(client)
 
-    @inlineCallbacks
-    def test_can_ping(self):
-        """Test that the client can send an active ping message and get a valid response."""
-        client = yield self.quick_register()
-        result = yield client.ping()
+    @pytest.mark.asyncio
+    async def test_can_moz_ping(self):
+        """Test that the client can send a small ping message and get a valid response."""
+        client = await self.quick_register()
+        result = await client.moz_ping()
         assert result == "{}"
-        assert client.ws.connected
+        assert client.ws.open
         try:
-            yield client.ping()
-        except AssertionError:
+            await client.moz_ping()
+        except websockets.exceptions.ConnectionClosedError:
             # pinging too quickly should disconnect without a valid ping
             # repsonse
             pass
-        assert not client.ws.connected
-        yield self.shut_down(client)
+        assert not client.ws.open
+        await self.shut_down(client)
 
-    @inlineCallbacks
-    def test_internal_endpoints(self):
+    @pytest.mark.asyncio
+    async def test_ws_ping(self):
+        """Test that the client can send a ping and get expected
+        future that returns the expected float latency value.
+        """
+        client = await self.quick_register()
+        result = await client.ping()
```

Comment:
I wasn't familiar with the [websockets ping API](https://websockets.readthedocs.io/en/stable/reference/asyncio/client.html#websockets.client.WebSocketClientProtocol.ping) so this confused me a bit: could we rename this `result` to `pong_waiter` like their example, and/or note in the method docstring we're awaiting a pong response.